http/middleware.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Middleware function based on {@linkcode Deno.ServeHandlerInfo} but an added
+ * `next()` function which calls the next middleware function in the middleware
+ * chain.
+ */
+export type MiddlewareFunction = (
+  request: Request,
+  info: Deno.ServeHandlerInfo,
+  next: () => Response | Promise<Response>,
+) => Response | Promise<Response>;
```

Comment:
Other possible function signatures like `(request, next)` and `(request, info)` have deliberately been omitted. I think having a plain function signature and just prefixing unused arguments in the middleware function with `_` is best.

Reply:
+1 on this decision. This is the right signature if we want middleware to be std-defined.